common/flatpak-dir.c

```diff
@@ -4018,19 +4022,231 @@ repo_get_remote_collection_id (OstreeRepo *repo,
   return TRUE;
 }
 
+static const SecretSchema *
+get_api_server_access_token_schema (void)
+{
+    static const SecretSchema schema = {
+        "com.endlessm.HackAccountToken", SECRET_SCHEMA_NONE,
+        {
+            {  "serial", SECRET_SCHEMA_ATTRIBUTE_STRING },
+            {  "NULL", 0 },
+        }
+    };
+    return &schema;
+}
+
+static char *
+get_api_server_access_token (GCancellable  *cancellable,
```

Comment:
As you say, the way to get the machine-specific credentials is pretty Hack-specific. I am not sure upstream would be happy with a mandatory libsecret dependency. You could just load the token from a well-known file in the user’s home directory, for now, and perhaps carry a patch to use libsecret downstream in Endless?

The design document isn’t particularly clear on the design for authenticating with the entitlement server, so I can’t really comment on what (if anything) would be more appropriate here.